### PR TITLE
Add TestException property to the TestContext

### LINKED ISSUE #293: How to get exception from test in TestCleanup?
We can get UnitTestOutcome.Error from TestContext.CurrentTestOutcome in TestCleanup. How to get additional information about error? May be Exception? It's can be important for cleanup.

### PR DESCRIPTION
**Pull request by Evangelink**
Fixes #293

### CODE REVIEW

File: src/Adapter/MSTestAdapter.PlatformServices/Interfaces/ITestContext.cs [hunk 1]
```diff
@@ -36,6 +36,12 @@ public interface ITestContext
     /// <param name="outcome"> The outcome. </param>
     void SetOutcome(UnitTestOutcome outcome);
 
+    /// <summary>
+    /// Sets the exception that occurred during the TestInitialize or TestMethod step (if any).
+    /// </summary>
+    /// <param name="exception">The exception.</param>
+    void SetException(Exception? exception);
```
Comment (Youssef1313) [inline src/Adapter/MSTestAdapter.PlatformServices/Interfaces/ITestContext.cs:43]: It seems like a breaking change. Not sure if okay.

File: test/IntegrationTests/MSTest.Acceptance.IntegrationTests/TestContextTests.cs [hunk 2]
```diff
@@ -34,6 +34,16 @@ public async Task TestContext_TestData_PropertyContainsExpectedValue()
...
     }
 
+    public async Task TestContext_TestException_PropertyContainsExpectedValue()
+    {
+        var testHost = TestHost.LocateFrom(_testAssetFixture.ProjectPath, TestAssetFixture.ProjectName, TargetFrameworks.NetCurrent.Arguments);
+        TestHostResult testHostResult = await testHost.ExecuteAsync("--filter ClassName~TestContextException");
+
+        // Assert
+        testHostResult.AssertExitCodeIs(2);
+        testHostResult.AssertOutputContainsSummary(failed: 2, passed: 1, skipped: 0);
```
Comment (Youssef1313) [inline test/IntegrationTests/MSTest.Acceptance.IntegrationTests/TestContextTests.cs:44]: Do we want to assert exactly that the two failures are `TestContextExceptionFailingInTestInit` and `TestContextExceptionFailingInTestMethod` while `TestContextExceptionNotFailing` passes?

File: src/Adapter/MSTestAdapter.PlatformServices/Interfaces/ITestContext.cs (hunk 1 above)
Comment (Evangelink) [reply src/Adapter/MSTestAdapter.PlatformServices/Interfaces/ITestContext.cs:43]: Good point! I couldn't find any ref to it using grepapp and as far as I can see in the codebase this is part of the types that are public for unknown reason as there is no way for user to provide a custom test context.

**Review by Youssef1313 — APPROVED** [review]

File: test/IntegrationTests/MSTest.Acceptance.IntegrationTests/TestContextTests.cs [hunk 3]
```diff
@@ -34,6 +34,18 @@ public async Task TestContext_TestData_PropertyContainsExpectedValue()
...
+        testHostResult.AssertOutputContains("Initialization method TestContextExceptionFailingInTestInit.TInit threw exception. System.InvalidOperationException");
+        testHostResult.AssertOutputContains("Test method TestContextExceptionFailingInTestMethod.TestFailingInTestMethod threw exception:");
```
Comment (Evangelink) [inline test/IntegrationTests/MSTest.Acceptance.IntegrationTests/TestContextTests.cs:46]: The assertion type is on the next line but I validated manually it's good.

**Pull request merged**